ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -107,13 +110,21 @@ public void stop(final QueryId queryId) {
     log.info("Attempting to stop Query: " + queryId.toString());
     if (collocatedQueries.containsKey(queryId) && sources.containsKey(queryId)) {
       if (kafkaStreams.state().isRunningOrRebalancing()) {
-        kafkaStreams.removeNamedTopology(queryId.toString());
+        try {
+          kafkaStreams.removeNamedTopology(queryId.toString(), true)
+              .all()
+              .get(30, TimeUnit.SECONDS);
```

Comment:
There's a ksql config for how long to wait for Streams to shutdown, ie for a query to terminate -- we should reuse that here. It's called `KSQL_SHUTDOWN_TIMEOUT_MS_CONFIG`